runtime/termscounter_test.go

```diff
@@ -0,0 +1,139 @@
+package runtime_test
+
+import (
+	"testing"
+	"time"
+
+	"go.etcd.io/gofail/runtime"
+)
+
+var __fp_ExampleString *runtime.Failpoint = runtime.NewFailpoint("ExampleString") //nolint:stylecheck
+
+func TestTermsCounter(t *testing.T) {
+	testcases := []struct {
+		name      string
+		fp        string
+		desc      string
+		runbefore int
+		runafter  int
+		want      int
+	}{
+		{
+			name:     "Terms limit Failpoint",
+			fp:       "ExampleString",
+			desc:     `10*sleep(10)->1*return("abc")`,
+			runafter: 12,
+			want:     11,
+		},
+		{
+			name:      "Inbetween Enabling Failpoint",
+			fp:        "ExampleString",
+			desc:      `10*sleep(10)->1*return("abc")`,
+			runbefore: 2,
+			runafter:  3,
+			want:      3,
+		},
+		{
+			name:      "Before Enabling Failpoint",
+			fp:        "ExampleString",
+			desc:      `10*sleep(10)->1*return("abc")`,
+			runbefore: 2,
+			runafter:  0,
+			want:      0,
+		},
+	}
+
+	for _, tc := range testcases {
+		t.Run(tc.name, func(t *testing.T) {
+
+			for i := 0; i < tc.runbefore; i++ {
+				exampleFunc()
+				time.Sleep(10 * time.Millisecond)
```

Comment:
Why sleep?

Reply:
I am not sure why this was included by the original author. I have removed it.